src/handler/mod.rs

```diff
@@ -0,0 +1,143 @@
+/// This module contains most of the low-level commands
+/// and grub variable modifications
+use anyhow::{bail, Result};
+use std::process::Command;
+use std::str;
+
+/// reboots the system if boot_counter is greater than 0 or can be forced too
+pub fn handle_reboot(force: bool) -> Result<()> {
+    if !force {
+        match get_boot_counter() {
+            Some(t) => {
+                if t <= 0 {
+                    bail!("countdown ended, check greenboot-rollback status")
+                }
+            }
+            None => bail!("boot_counter is not set"),
+        }
+    }
+    log::info!("restarting system");
+    let status = Command::new("systemctl").arg("reboot").status()?;
+    if status.success() {
+        return Ok(());
+    }
+    bail!("systemd returned error");
```

Comment:
We need to return the systemd error here.

Reply:
we should never reach here but I agree we need a return err